swan-lake/resources/featured-scenarios/write-a-restful-api-with-ballerina.md

```diff
@@ -28,76 +26,161 @@ This guide will walk you through creating a RESTful API with two endpoints. In t
 
 > **Info:** For more information on Ballerina packages, see [Organize Ballerina code](/learn/organize-ballerina-code/).
 
-```
-$ bal new covid19 -t service
-```
+1. In the terminal, execute the command below to create the Ballerina package for the API implementation.
 
-You view the output below.
+    ```
+    $ bal new write-a-restful-api
+    ```
 
-```
-Created new package 'covid19' at covid19.
-```
+    You should see the output similar to the following.
 
-This creates a directory named `covid19` with the default module along with a sample code for the service as shown below. 
+    ```
+    package name is derived as 'write_a_restful_api'. Edit the Ballerina.toml to change it.
 
-```
-.
-├── covid19
-│   ├── Ballerina.toml
-│   └── service.bal
-```
+    Created new package 'write_a_restful_api' at /Users/write-a-restful-api.
+    ```
+
+    This creates a directory named `write-a-restful-api` with sample Ballerina code, as shown below. 
+
+    ```
+    .
+    ├── Ballerina.toml
+    └── main.bal
+    ```
+
+    - `Ballerina.toml` is the file that makes the folder a Ballerina package. 
+    - The `main.bal` file provides the look and feel of a simple Ballerina program. 
+
+2. Open the created package in VS Code.
 
-- `Ballerina.toml` is the file that makes the folder a Ballerina package. It also contains a test directory to include tests for the service. However, this will not be used in this guide. 
-- The `service.bal` template file provides a look and feel about Ballerina services. 
+3. Remove the generated content in the `main.bal` file and open the diagram view in VS Code.
+
+    ![Open diagram view](/learn/images/featured-scenarios/write-a-restful-api-with-ballerina/open-diagram-view.gif)
 
 ## Create the dataset
 
-To keep things simple, an in-memory dataset is used with three entries. To add the definition of the record and the declaration of the table, replace the API template file (i.e., `service.bal`) with the code below.
+An in-memory dataset with three entries is used to keep things simple. Follow the steps below to add the definition of the record and the declaration of the [table](/learn/by-example/table/) that holds the data.
+
+1. Generate the record types corresponding to the payload of the REST service by providing the record name as `CovidEntry` and the sample JSON object below.
+
+    >**Tip:** You need to complete the generated record by adding the pipe signs to mark the record as a closed one and adding the `readonly` descriptor to the `iso_code` field to make it non-modifiable as it is the key of the table, which cannot be represented in the JSON format.
+
+    ```json
+    {
+        "iso_code": "AFG",
+        "country": "Afghanistan",
+        "cases": 159303,
+        "deaths": 7386,
+        "recovered": 146084,
+        "active": 5833
+    }
+    ```
+    ![Create data record](/learn/images/featured-scenarios/write-a-restful-api-with-ballerina/create-data-record.gif)
+
```

Comment:
With integration tutorials, we also added the generated code after a step like this.

https://ballerina.io/learn/integration-tutorials/service-orchestration/#step-2-develop-the-service

Reply:
Updated.